File: docs/vsto/codesnippet/CSharp/Trin_WordAddInDynamicControlsWalkthrough/MyRibbon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Office.Tools.Ribbon;

namespace Trin_WordAddInDynamicControlsWalkthrough
{
    public partial class MyRibbon : RibbonBase
    {
        public MyRibbon() : base(Globals.Factory.GetRibbonFactory())
        {
            InitializeComponent();
        }

        private void MyRibbon_Load(object sender, RibbonUIEventArgs e)
        {

        }

        //<Snippet6>
        private void addButtonCheckBox_Click(object sender, RibbonControlEventArgs e)
        {
            Globals.ThisAddIn.ToggleButtonOnDocument();
        }

        private void addRichTextCheckBox_Click(object sender, RibbonControlEventArgs e)
        {
            Globals.ThisAddIn.ToggleRichTextControlOnDocument();
        }
        //</Snippet6>
    }
}

File: docs/vsto/codesnippet/VisualBasic/Trin_WordAddInDynamicControlsWalkthrough/MyRibbon.vb
Imports Microsoft.Office.Tools.Ribbon

Public Class MyRibbon

    Private Sub MyRibbon_Load(ByVal sender As System.Object, ByVal e As RibbonUIEventArgs) Handles MyBase.Load

    End Sub

    '<Snippet6>
    Private Sub addButtonCheckBox_Click(ByVal sender As System.Object, _
        ByVal e As RibbonControlEventArgs) Handles addButtonCheckBox.Click
        Globals.ThisAddIn.ToggleButtonOnDocument()
    End Sub

    Private Sub addRichTextCheckBox_Click(ByVal sender As System.Object, _
        ByVal e As RibbonControlEventArgs) Handles addRichTextCheckBox.Click
        Globals.ThisAddIn.ToggleRichTextControlOnDocument()
    End Sub
    '</Snippet6>
End Class
